pkg/scheduler/core/spreadconstraint/select_groups.go

```diff
@@ -161,9 +161,11 @@ func findFeasiblePaths(groups []*GroupInfo, minConstraint, maxConstraint, target
 	rootPath := new(dfsPath)
 	var dfsFunc func(int, int)
 	dfsFunc = func(sum, begin int) {
-		if sum >= target && rootPath.length() >= minConstraint && rootPath.length() <= maxConstraint {
-			paths = append(paths, rootPath.next())
-			return
+		if sum >= target && rootPath.length() >= minConstraint {
+			if maxConstraint == 0 || rootPath.length() <= maxConstraint {
```

Comment:
The modifications here may be more sensitive. In fact, the real reason why no cluster is selected is that the function ends here.
```go
                // pruning makes DFS faster
		if rootPath.length() >= maxConstraint {
			return
		}
```
I prefer to set `maxGroup` to an infinite value, such as `math.MaxInt`, to avoid modifying these codes.